.github/workflows/ci.yml

```diff
@@ -155,69 +104,31 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Setup PHP version
+        id: setup-php
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ env.PHP_VERSION }}
           coverage: none
 
-      - name: Pull composer vendor cache
-        uses: actions/cache/restore@v3
-        with:
-          path: vendor
-          fail-on-cache-miss: true
-          key: composer-build-dev-${{ hashFiles('**/composer.lock') }}
-
-      - name: Regenerate autoload file # If autoloading paths were added before the files themselves, the autoload cache is out of date
-        run: COMPOSER_HOME=$GLOBAL_COMPOSER_CACHE_PATH composer dump-autoload
+      - name: Install dependencies
+        uses: ramsey/composer-install@v2
 
-      - name: Pull PHPStan cache
-        id: pull-phpstan-cache
-        uses: actions/cache/restore@v3
+      - name: Cache PHPStan
+        uses: actions/cache@v3
         with:
           path: ${{ env.PHPSTAN_CACHE_DIR }}
-          key: phpstan-${{ github.ref_name }}
-          restore-keys: |
-            phpstan
+          key: ${{ steps.setup-php.outputs.php-version }}-phpstan-${{ hashFiles('phpstan.neon') }}
+          restore-keys: ${{ steps.setup-php.outputs.php-version }}-phpstan-
 
       - name: Run PHPStan
-        run: composer stan
-
-      - name: Install gh-actions cache plugin
-        if: always()
-        run: gh extension install actions/gh-actions-cache
-
-      - name: Clear previous version of PHPStan branch cache
-        if: always() # We always delete the cache, because even if it was not 'hit', it might have been created by a commit in the meantime
-        continue-on-error: true
-        run: gh actions-cache delete phpstan-${{ github.ref_name }} -B ${{ github.ref }} --confirm
-
-      - name: Push PHPStan branch cache
```

Comment:
This mechanism exists so that when a PHPStan cache on master is built, it won't stay there for 90 days getting more outdated with each change. So we pull the cache, and delete it and reupload it after the job has finished, irregardless of if it was successfull.